File: Bai_34/main.py
import math

def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

with open('E:\\Project Visual Studio Code\\Summary_exercises\\Bai_34\\NTMAX.INP', 'r') as infile:
    T = infile.readline().strip()
    digit_count = 0
    max_prime = 0
    current_number = ""

    for c in T:
        if c.isdigit():
            digit_count += 1
            current_number += c
        else:
            if current_number:
                number = int(current_number)
                if is_prime(number) and number > max_prime:
                    max_prime = number
                current_number = ""

    if current_number:
        number = int(current_number)
        if is_prime(number) and number > max_prime:
            max_prime = number

with open('E:\\Project Visual Studio Code\\Summary_exercises\\Bai_34\\NTMAX.OUT', 'w') as outfile:
    outfile.write(str(digit_count) + '\n')
    outfile.write(str(max_prime) + '\n')

File: Bai_34/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <cmath>

using namespace std;

bool is_prime(int n) {
    if (n < 2) return false;
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) return false;
    }
    return true;
}

int main() {
    // Mở file đầu vào NTMAX.INP
    ifstream infile("E:\\Project Visual Studio Code\\Summary_exercises\\Bai_34\\NTMAX.INP");
    // Mở file đầu ra NTMAX.OUT
    ofstream outfile("E:\\Project Visual Studio Code\\Summary_exercises\\Bai_34\\NTMAX.OUT");
    
    string T;
    getline(infile, T); // Đọc xâu từ file

    infile.close(); // Đóng file sau khi đọc xong

    // Các bước tiếp theo giống như trong mã nguồn trước

    int digit_count = 0;
    int max_prime = 0;
    string current_number = "";

    for(char c : T) 
    {
        if(isdigit(c)) 
        {
            digit_count++;
            current_number += c;
        } 
        else 
        {
            if(!current_number.empty()) 
            {
                int number = stoi(current_number);
                if (is_prime(number) && number > max_prime) 
                    max_prime = number;

                current_number = "";
            }
        }
    }

    if (!current_number.empty()) 
    {
        int number = stoi(current_number);
        if (is_prime(number) && number > max_prime) 
            max_prime = number;
    }

    // In kết quả ra file
    outfile << digit_count << endl;
    outfile << max_prime << endl;

    // Đóng file đầu ra
    outfile.close();

    return 0;
}
